homeassistant/components/growatt_server/sensor.py

```diff
@@ -105,6 +103,11 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     add_entities(entities, True)
 
 
+async def async_setup_entry(hass, config, add_entities):
+    """Set up growatt server from Config Flow."""
+    setup_platform(hass, config.data, add_entities)
```

Comment:
We can't mix async context with sync context. Here were in async context and we want to call a function that should run in sync context, ie in the executor thread pool. Then we need to schedule the target to run in the thread pool with `hass.async_add_executor_job`.